scipy/optimize/tests/test_optimize.py

```diff
@@ -1127,6 +1127,8 @@ def test_minimize_callback_copies_array(self, method):
                 kwargs['iter'] = 5
             elif method == 'fmin_tnc':
                 kwargs['maxfun'] = 100
+            elif method in ('fmin', 'fmin_powell'):
+                kwargs['maxiter'] = 3500
```

Comment:
I'd suggest only changing test warnings to `Warning` objects in this PR rather than adding new warnings.

Reply:
I'm sorry. Could you elaborate it?